bool is_swap
float32[] rewards_rm
float32[] rewards_dv
int32[] slot_states
int32[] reachable_slots
